core/src/main/scala/org/http4s/Query.scala

```diff
@@ -17,28 +17,51 @@ import scala.collection.mutable.ListBuffer
   * When rendered, the resulting `String` will have the pairs separated
   * by '&' while the key is separated from the value with '='
   */
-final class Query private (pairs: Vector[KeyValue])
-    extends IndexedSeq[KeyValue]
-    with IndexedSeqOptimized[KeyValue, Query]
-    with QueryOps
-    with Renderable {
-  override def apply(idx: Int): KeyValue = pairs(idx)
+final class Query private (val pairs: Vector[KeyValue]) extends QueryOps with Renderable {
+  // override def apply(idx: Int): KeyValue = pairs(idx)
```

Comment:
Does this need a `@deprecated` or should it just be removed?

Reply:
I see no reason we can't keep this, though it suggests random access in the implementation. I'm ambivalent about deprecating it or not.